gpt_engineer/applications/cli/main.py

```diff
@@ -0,0 +1,216 @@
+"""
+This module provides a CLI tool to interact with the GPT Engineer application,
+enabling users to use OpenAI's models and define various parameters for the
+project they want to generate, improve or interact with.
+
+Main Functionality:
+---------------------
+- Load environment variables needed to work with OpenAI.
+- Allow users to specify parameters such as:
+  - Project path
+  - Model type (default to GPT-4)
+  - Temperature
+  - Step configurations
+  - Code improvement mode
+  - Lite mode for lighter operations
+  - Azure endpoint for Azure OpenAI services
+  - Using project's preprompts or default ones
+  - Verbosity level for logging
+- Interact with AI, databases, and archive processes based on the user-defined parameters.
+
+Notes:
+- Ensure the .env file has the `OPENAI_API_KEY` or provide it in the working directory.
+- The default project path is set to `projects/example`.
+- For azure_endpoint, provide the endpoint for Azure OpenAI service.
+
+"""
+
+import os
+from pathlib import Path
+
+import openai
+import typer
+from dotenv import load_dotenv
+
+from gpt_engineer.core.default.on_disk_repository import OnDiskRepository
+from gpt_engineer.core.ai import AI
+from gpt_engineer.core.default.paths import PREPROMPTS_PATH
+from gpt_engineer.applications.cli.file_selector import ask_for_files, get_all_code
+from gpt_engineer.tools.custom_steps import (
+    lite_gen,
+    clarified_gen,
+    self_heal,
+    vector_improve,
+)
+from gpt_engineer.core.default.git_version_manager import GitVersionManager
+from gpt_engineer.core.default.steps import gen_code, execute_entrypoint, improve
+from gpt_engineer.applications.cli.cli_agent import CliAgent
+from gpt_engineer.applications.cli.collect import collect_and_send_human_review
+from gpt_engineer.core.preprompts_holder import PrepromptsHolder
+import logging
+
+app = typer.Typer()  # creates a CLI app
+
+
+def load_env_if_needed():
+    if os.getenv("OPENAI_API_KEY") is None:
+        load_dotenv()
+    if os.getenv("OPENAI_API_KEY") is None:
+        # if there is no .env file, try to load from the current working directory
+        load_dotenv(dotenv_path=os.path.join(os.getcwd(), ".env"))
+    openai.api_key = os.getenv("OPENAI_API_KEY")
```

Comment:
what if there is still not api key? we should notify the user.

Reply:
I'm not really sure how you mean. If no key is provided, an openai.error.AuthenticationError is raised with a failure message. We can of course catch this, etc, but is this not informative enough?

Reply:
I would catch that case and throw an Error with a more descriptive error message with a suggested action. This is optional though.